pkg/jobmanager/jobmanager.go

```diff
@@ -45,3 +55,69 @@ func (jm *JobManager) LookupJob(id uuid.UUID) (*Job, error) {
 	}
 	return j, nil
 }
+
+func (jm *JobManager) StopJob(id uuid.UUID) error {
+	j, err := jm.LookupJob(id)
+	if err != nil {
+		return err
+	}
+	j.mu.Lock() // NOTE: We don't defer unlock as we want to wait after unlock unpon success.
+	if j.status != pb.JobStatus_JOB_STATUS_RUNNING {
+		j.mu.Unlock()
+		return nil
+	}
+	if j.cmd.Process != nil {
+		// Send the KILL to the process group, not just the process to avoid orphans.
+		// NOTE: This is POSIX compliant.
+		if err := syscall.Kill(-j.cmd.Process.Pid, syscall.SIGKILL); err != nil {
```

Comment:
Doesn't this race with the waitpid in the `exec.Cmd.Wait` running as part of the backgrounded `Job.wait`?

Reply:
Yes, it does. I describe this in the comment in the error block. It is not a 'data race' though, the go race detector cannot detect it, and it is un-avoidable (unless I missed something)

Reply:
Added TODO referring to #9